src/irmin-pack/inode.ml

```diff
@@ -755,6 +755,208 @@ struct
       in
       Irmin.Type.map I.t ~pre_hash (fun v -> { find = niet; v }) (fun t -> t.v)
 
+    module type DATA_FORMAT = sig
+      type vt
+      type t
+
+      exception Misconstructed_Data of string
+
+      val of_int : ?label:string -> int -> t
+      val to_int : ?label:string -> t -> int
+      val of_string : ?label:string -> string -> t
+      val to_string : ?label:string -> t -> string
+      val of_hash : ?label:string -> hash -> t
+      val to_hash : ?label:string -> t -> hash
+      val of_lazy_hash : ?label:string -> hash lazy_t -> t
+      val to_lazy_hash : ?label:string -> t -> hash lazy_t
+      val of_metadata : ?label:string -> metadata -> t
+      val to_metadata : ?label:string -> t -> metadata
+
+      val of_value :
+        ?label:string ->
+        ?label_hash:string ->
+        ?label_metadata:string ->
+        [< `Contents of hash * metadata | `Node of hash ] ->
+        t
+
+      val to_value :
+        ?label:string ->
+        ?label_hash:string ->
+        ?label_metadata:string ->
+        t ->
+        [> `Contents of hash * metadata | `Node of hash ]
+
+      val of_step : ?label:string -> step -> t
+      val to_step : ?label:string -> t -> step
+      val join : ?label:string -> t list -> t
+      val disjoin : ?label:string -> t -> t list
+      val parse_from_file : string -> t
```

Comment:
```suggestion
```

Convenient, but let's avoid introducing IO dependencies where they're not strictly necessary – in a future project, we'll need to functorise over IO.

Reply:
No problem, I put it here just in case :-)